docs/UPGRADE_GUIDE.md

```diff
@@ -1,6 +1,40 @@
 # Upgrade Guide
 
-## Migrating from <=v0.8 to 0.9
+## Migrating from <=v0.23 to v0.24
+
+Prior to v0.24, the Deletion Queue was synchronously processed on Job Creation
+and stored in DynamoDB. As a result, the job API provided the full queue for a
+given job in the `DeletionQueueItems` property and there was a limit of ~375KB
+on the queue size for each individual Job (if the size for a given job would
```

Comment:
nit: don't think the second statement needs to be in brackets. It's a separate sentence